tracer/src/Datadog.Trace/Vendors/IndieSystem.Text.RegularExpressions/System/Threading/StackHelper.cs

```diff
@@ -20,22 +20,11 @@ internal static class StackHelper
         /// <summary>Tries to ensure there is sufficient stack to execute the average .NET function.</summary>
         public static bool TryEnsureSufficientExecutionStack()
         {
-#if REGEXGENERATOR
-            try
-            {
-                RuntimeHelpers.EnsureSufficientExecutionStack();
-                return true;
-            }
-            catch
-            {
-                return false;
-            }
-#else
```

Comment:
You wrapped `RuntimeHelpers.EnsureSufficientExecutionStack` in a try-catch here, but below it doesn't have one... do we need a try catch around the latter?

Reply:
I set `REGEXGENERATOR` in the last PR because to make some types internal, as the optional compilation meant they didn't match the regex which should change them.

This was the wrong thing to do, since `REGEXGENERATOR` only be used in the regex source generator, which we're not using, so cleaning it out will make the codes behaviour closer to that of the actual library.